client/protocol.py

```diff
@@ -0,0 +1,200 @@
+
+        ok_cnt = 0
+        is_received_leader_res = False
+        is_received_success_res = None
+        cmd_res = CommandResponse()
+        exe_err = ExecuteError(_ExecuteError())
+        propose_err = ProposeError(_ProposeError())
+
+        for stub in self.inner:
+            res = await propose_wrapper(stub, cmd)
+
+            if res.HasField("result"):
+                cmd_result = res.result
+                ok_cnt += 1
+                is_received_leader_res = True
+                if cmd_result.HasField("er"):
+                    is_received_success_res = True
+                    cmd_res.ParseFromString(cmd_result.er)
+                if cmd_result.HasField("error"):
+                    is_received_success_res = False
+                    exe_err.inner.ParseFromString(cmd_result.error)
+            elif res.HasField("error"):
+                is_received_leader_res = True
+                propose_err.inner.CopyFrom(res.error)
+            else:
+                ok_cnt += 1
+
+            if is_received_leader_res and ok_cnt >= super_quorum(len(self.connects)):
+                logging.info("fast round success")
+                if is_received_success_res:
+                    return cmd_res
+                if not is_received_success_res:
+                    raise exe_err
+
+        raise propose_err
+
+    async def slow_round(self, cmd: Command) -> tuple[SyncResponse, CommandResponse]:
+        """The slow round of Curp protocol"""
+
+        logging.info("slow round start. propose id: %s", cmd.propose_id)
+
+        addr = ""
+        sync_res = SyncResponse()
+        cmd_res = CommandResponse()
+        exe_err = CommandSyncError(_CommandSyncError())
+        after_sync_err = WaitSyncError(_WaitSyncError())
+
+        for member in self.connects:
+            if member.id == self.local_server_id:
+                addr = member.name
+                break
+
+        channel = grpc.insecure_channel(addr)
+        stub = ProtocolStub(channel)
+        res = await wait_synced_wrapper(stub, cmd)
+
+        if res.HasField("success"):
+            success = res.success
+            sync_res.ParseFromString(success.after_sync_result)
+            cmd_res.ParseFromString(success.exe_result)
+            return (sync_res, cmd_res)
+        if res.HasField("error"):
+            cmd_sync_err = res.error
+            if cmd_sync_err.HasField("execute"):
+                exe_err.inner.ParseFromString(cmd_sync_err.execute)
+                raise exe_err
+            if cmd_sync_err.HasField("after_sync"):
+                after_sync_err.inner.ParseFromString(cmd_sync_err.after_sync)
+                raise after_sync_err
+
+        err_msg = "Response decode error"
+        raise ResDecodeError(err_msg)
+
+
+def fetch_cluster(stubs: list[ProtocolStub]) -> FetchClusterResponse:
+    """Fetch cluster from server"""
+    for stub in stubs:
+        res: FetchClusterResponse = stub.FetchCluster(FetchClusterRequest())
+    return res
```

Comment:
This function seems unimplemented. Please add a TODO if it's going to be implemented later.